salt/tests/unit/modules/test_metalk8s_checks.py

```diff
@@ -180,6 +183,40 @@ def test_sysctl(self, params, data, result, raises=False):
             else:
                 self.assertEqual(metalk8s_checks.sysctl(params, raises=False), result)
 
+    @utils.parameterized_from_cases(YAML_TESTS_CASES["containerd_filesystem"])
+    def test_containerd_filesystem(
+        self,
+        result,
+        run_all_df_ret=None,
+        xfs_info_ret=None,
+        expect_raise=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `containerd_filesystem` function
+        """
+        run_all_df_mock = MagicMock(return_value=run_all_df_ret)
+        xfs_info_mock = MagicMock()
+        if xfs_info_ret and 'exception' in xfs_info_ret:
+            xfs_info_mock.side_effect = CommandExecutionError(
+                xfs_info_ret["exception"]
+            )
+        else:
+            xfs_info_mock.return_value = xfs_info_ret
+
+        patch_dict = {
+            "cmd.run_all": run_all_df_mock,
+            "xfs.info": xfs_info_mock
+        }
+
+        with patch.dict(metalk8s_checks.__salt__, patch_dict):
```

Comment:
I think you want to mock and test the `os.path.exists` otherwise I think if `/var/lib/containerd` (and `/foo/bar`) directory exists when you run the test you do not get 100% coverage
Which is sad that test coverage depends on the environment where you run the test :smile: